ember-toucan-core/src/components/form/controls/checkbox.ts

```diff
@@ -0,0 +1,74 @@
+import Component from '@glimmer/component';
+import { assert } from '@ember/debug';
+import { action } from '@ember/object';
+
+export interface ToucanFormCheckboxControlComponentSignature {
+  Element: HTMLInputElement;
+  Args: {
+    isDisabled?: boolean;
+    onChange?: (
+      isChecked: boolean,
+      e: Event | InputEvent,
+      isIndeterminate: boolean
+    ) => void;
+    isIndeterminate?: boolean;
+    value?: boolean;
+  };
+}
+
+export default class ToucanFormCheckboxControlComponent extends Component<ToucanFormCheckboxControlComponentSignature> {
+  get isChecked() {
+    return this.args.value ?? false;
+  }
+
+  get isIndeterminate() {
+    return this.args.isIndeterminate ?? false;
+  }
+
+  /**
+   * Used in our template so we don't need to pull in ember-truth-helpers (yet).
+   */
+  get isCheckedOrIndeterminate() {
+    return this.isChecked || this.isIndeterminate;
+  }
+
+  /**
+   * We use the data-checked attribute to help signal what state the checkbox is in.
+   * This is most helpful in tests. The "mixed" value is used for the indeterminate state.
+   * We don't use aria-checked directly, as we are a valid <input type="checkbox", but
+   * we follow the possible states outlined by aria-checked.
+   *
+   * @see https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-checked
+   */
+  get checkedState() {
```

Comment:
I am not entirely sure about why we need this? 

You could check all possible states in tests using `isChecked()` or `hasProperty('indeterminate', true)`, right?

Reply:
Yeah maybe that makes more sense.  What do you think?  My initial reaction was to provide something to make it easy to check states in tests, but maybe instead we write some documentation suggesting what you have here instead?

Reply:
Went ahead and went this route!